File: backend/src/models/Attendance.js
const mongoose = require("mongoose");

// CHAM_CONG: cong nhan tu bam vao/ra hang ngay, chi de theo doi/doi chieu, khong anh huong tinh luong
const attendanceSchema = new mongoose.Schema(
  {
    worker: { type: mongoose.Schema.Types.ObjectId, ref: "Worker", required: true },
    date: { type: Date, required: true }, // ngay cong, chuan ve 00:00 gio server
    checkInAt: { type: Date },
    checkOutAt: { type: Date },
  },
  { timestamps: true }
);

attendanceSchema.index({ worker: 1, date: 1 }, { unique: true });

module.exports = mongoose.model("Attendance", attendanceSchema);

File: app/lib/models/attendance.dart
class Attendance {
  final String id;
  final DateTime date;
  final DateTime? checkInAt;
  final DateTime? checkOutAt;

  Attendance({
    required this.id,
    required this.date,
    this.checkInAt,
    this.checkOutAt,
  });

  factory Attendance.fromJson(Map<String, dynamic> json) => Attendance(
        id: json['_id'] as String,
        date: DateTime.parse(json['date'] as String),
        checkInAt: json['checkInAt'] != null ? DateTime.parse(json['checkInAt'] as String) : null,
        checkOutAt: json['checkOutAt'] != null ? DateTime.parse(json['checkOutAt'] as String) : null,
      );
}
